src/main/scala/epic/dense/AffineTransform.scala

```diff
@@ -78,6 +108,44 @@ object AffineTransform {
                                                                     canAxpy: scaleAdd.InPlaceImpl3[DenseVector[Double], Double, FV]) = new AffineTransform(numOutputs, numInputs, new IdentityTransform[FV], includeBias)
   def apply(numOutputs: Int, numInputs: Int, includeBias: Boolean):AffineTransform[DenseVector[Double], DenseVector[Double]] = apply(numOutputs, numInputs, new IdentityTransform[DenseVector[Double]], includeBias)
   def apply(numOutputs: Int, numInputs: Int):AffineTransform[DenseVector[Double], DenseVector[Double]]  = apply(numOutputs, numInputs, true)
+  
+  
+  def getUniformAffineWeights(numWeights: Int, initWeightsScale: Double, rng: Random) = {
+    DenseVector(Array.tabulate(numWeights)(i => rng.nextGaussian * initWeightsScale))
+  }
+  
+  def getGaussianAffineWeights(numWeights: Int, initWeightsScale: Double, rng: Random) = {
+    DenseVector(Array.tabulate(numWeights)(i => rng.nextGaussian * initWeightsScale))
+  }
+  
+  // N.B. numWeights != inSize * outSize if there's a bias
+  def getMagicAffineWeights(numWeights: Int, inSize: Int, outSize: Int, initWeightsScale: Double, rng: Random) = {
```

Comment:
This is the initialization for weights in hidden (+ input) layers, right? A value drawn uniformly at random from the interval [- √( 6 /  # layer units ), √( 6 /  # layer units )] ?